### PR TITLE
Refactor syntax unicode iphone reachability

### PR DESCRIPTION
**Pull request by livecodesebastien**

### CODE REVIEW

File: engine/src/mbliphonereachability.mm [hunk 1]
```diff
@@ -60,11 +60,11 @@
...
 };
 
-static bool MCAddReachabilityTarget(const char *p_host);
-static bool MCRemoveReachabilityTarget(const char *p_host);
+static bool MCAddReachabilityTarget(MCStringRef p_host);
+static bool MCRemoveReachabilityTarget(MCStringRef p_host);
 
 static SCNetworkReachabilityRef s_reach_ref = nil;
-char *s_reach_target = nil;
+MCAutoStringRef s_reach_target;
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: Globals mustn't be MCAutoStringRef... They have to be MCStringRef. Such things should be initialized/finalized in an Initialize/Finalize pair (MCReachabilityInitialize / MCReachabilityFinalize). These init/finit methods should be called from X_clear_globals / X_close.

File: engine/src/mbliphonereachability.mm [hunk 2]
```diff
@@ -145,57 +145,51 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
 	SCNetworkReachabilityRef t_reach = nil;
 	SCNetworkReachabilityContext t_context;
 	
-	char *t_target = nil;
-	
-	if (t_success)
-		t_success = nil != (t_reach = SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, p_host));
 	if (t_success)
-		t_success = MCCStringClone(p_host, t_target);
+        t_success = nil != (t_reach = SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, MCStringGetCString(p_host)));
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: As this API call requires a CString, you need to convert the StringRef to one - use MCStringConvertToCString.

File: engine/src/mbliphonereachability.mm [hunk 3]
```diff
@@ -145,57 +145,51 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
 	if (t_success)
 	{
 		t_context.version = 0;
 		t_context.copyDescription = nil;
 		t_context.retain = nil;
 		t_context.release = nil;
-		t_context.info = t_target;
+        t_context.info = (void*)MCStringGetCString(p_host);
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: Here t_context.info is a pointer which persists after this method finishes - so you can use the c-string created with Convert here (i.e. no need to free it unless an error occurs).

File: engine/src/mbliphonereachability.mm [hunk 4]
```diff
@@ -145,57 +145,51 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
 	{
 		SCNetworkReachabilityScheduleWithRunLoop(t_reach, CFRunLoopGetCurrent(), kCFRunLoopDefaultMode);
-		if (s_reach_target != nil)
-			MCRemoveReachabilityTarget(s_reach_target);
+        if (*s_reach_target != nil)
+            MCRemoveReachabilityTarget(*s_reach_target);
 		
 		s_reach_ref = t_reach;
-		s_reach_target = t_target;
+        /* UNCHECKED */ MCStringCopy(p_host, &s_reach_target);
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: Use MCStringAssign(s_reach_target, p_host) here. We should use the default convention that s_reach_target is initialized to kMCEmptyString.

File: engine/src/mbliphonereachability.mm (hunk 1 above)
Comment (livecodesebastien) [inline engine/src/mbliphonereachability.mm]: I was quite unsure this would be correct, but didn't think about adding the Init/Finalize functions in X_clear_globals / X_close

File: engine/src/mbliphonereachability.mm [hunk 5]
```diff
@@ -217,8 +211,8 @@ bool MCSystemGetReachabilityTarget(MCStringRef& r_reach_target)
 	MCresult -> copysvalue(MCReachabilityGetTarget());
 	return ES_NORMAL;
 #endif /* MCHandleReachabilityTarget */
-    if (s_reach_target != nil)
-        return MCStringCreateWithCString(s_reach_target, r_reach_target);
+    if (*s_reach_target != nil)
+        return MCStringCopy(*s_reach_target, r_reach_target);
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: Don't use MCStringCopy - just r_reach_target = MCValueRetain(s_reach_target). (also with the init to empty string, no need to test for a nil ptr).

File: engine/src/mbliphonereachability.mm [hunk 6]
```diff
@@ -145,70 +147,75 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
 	MCEventQueuePostCustom(t_event);
 }
 
-static bool MCAddReachabilityTarget(const char *p_host)
+static bool MCAddReachabilityTarget(MCStringRef p_host)
 {
 	bool t_success = true;
 	SCNetworkReachabilityRef t_reach = nil;
 	SCNetworkReachabilityContext t_context;
+    char* t_host_cstring;
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: Initialize this to nil before doing the Convert.

File: engine/src/mbliphonereachability.mm [hunk 7]
```diff
@@ -145,70 +147,75 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
+    
+    /* UNECHECKED */ MCStringConvertToCString(p_host, t_host_cstring);
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm]: I think you mean /\* UNCHECKED */

File: engine/src/mbliphonereachability.mm [hunk 8]
```diff
@@ -145,70 +147,75 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
-		s_reach_target = t_target;
+        s_reach_target = MCValueRetain(p_host);
 	}
 	else
 	{
 		if (t_reach != nil)
-			CFRelease(t_reach);
-		if (t_target != nil)
-			MCCStringFree(t_target);
+            CFRelease(t_reach);
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm:188]: If the call fails, you need to free t_host_cstring.

File: engine/src/mbliphonereachability.mm [hunk 9]
```diff
@@ -145,70 +147,75 @@ static void reachability_callback(SCNetworkReachabilityRef p_target, SCNetworkRe
...
-	char *t_target = nil;
-	
-	if (t_success)
-		t_success = nil != (t_reach = SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, p_host));
 	if (t_success)
-		t_success = MCCStringClone(p_host, t_target);
+        t_success = nil != (t_reach = SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, t_host_cstring));
 	if (t_success)
 	{
 		t_context.version = 0;
```
Comment (runrevmark) [inline engine/src/mbliphonereachability.mm:170]: Actually, looking at the API for the SetCallback method - the retain / release fields are used to manage the lifetime of the info field... Therefore, I suggest making the info field p_host (i.e. an MCStringRef), and then setting retain/release to MCValueRetain / MCValueRelease. This means we don't have to worry about freeing the info data anywhere. It also means t_host_cstring can be auto-managed with an MCAutoPointer.

Comment (runrevmark) [inline engine/src/mbliphonereachability.mm:170]: This change the ReachabilityEvent will need to be updated so it treates info as a MCStringRef, rather than a cstring.
